pkg/provisioningv2/rke2/planner/config.go

```diff
@@ -413,6 +435,72 @@ func configFile(controlPlane *rkev1.RKEControlPlane, filename string) string {
 		rke2.GetRuntime(controlPlane.Spec.KubernetesVersion), filename)
 }
 
+func (p *Planner) renderFiles(controlPlane *rkev1.RKEControlPlane, entry *planEntry) ([]plan.File, error) {
+	var files []plan.File
+	for _, msf := range controlPlane.Spec.MachineSelectorFiles {
+		sel, err := metav1.LabelSelectorAsSelector(msf.MachineLabelSelector)
+		if err != nil {
+			return files, err
+		}
+		if msf.MachineLabelSelector == nil || sel.Matches(labels.Set(entry.Machine.Labels)) {
```

Comment:
Nit: We could invert some if statements and break this out into a function that operates on `K8sObjectFileSource` to bring the indentation down, if you're into that sort of thing. I won't prevent from merging for this though.

Reply:
I'm not going to break out the function just yet, but I did invert the if statement